src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.State.cs

```diff
@@ -208,20 +166,34 @@ private sealed class FullDeclarationState : State
                     : base(new WeakValueSource<Compilation>(declarationCompilation),
                            declarationCompilation.Clone().RemoveAllReferences(),
                            generatedDocuments,
-                           generatedDocumentsAreFinal,
-                           GetUnrootedSymbols(declarationCompilation))
+                           generatedDocumentsAreFinal)
                 {
                 }
             }
 
             /// <summary>
-            /// The final state a compilation tracker reaches. The <see cref="State.DeclarationOnlyCompilation"/> is available,
-            /// as well as the real <see cref="State.FinalCompilationWithGeneratedDocuments"/>.
+            /// The final state a compilation tracker reaches. The <see cref="State.DeclarationOnlyCompilation"/> is
+            /// available, as well as the real <see cref="State.FinalCompilationWithGeneratedDocuments"/>. It is a
+            /// requirement that any <see cref="Compilation"/> provided to any clients of the <see cref="Solution"/>
+            /// (for example, through <see cref="Project.GetCompilationAsync"/> or <see
+            /// cref="Project.TryGetCompilation"/> must be from a <see cref="FinalState"/>.  This is because <see
+            /// cref="FinalState"/> stores extra information in it about that compilation that the <see
+            /// cref="Solution"/> can be queried for (for example: <see
+            /// cref="Solution.GetOriginatingProject(ISymbol)"/>.  If <see cref="Compilation"/>s from other <see
+            /// cref="State"/>s are passed out, then these other APIs will not function correctly.
             /// </summary>
             private sealed class FinalState : State
             {
                 public override bool? HasSuccessfullyLoaded { get; }
 
+                /// <summary>
+                /// Weak set of the assembly, module and dynamic symbols that this compilation tracker has created.
+                /// This can be used to determine which project an assembly symbol came from after the fact.  This is
+                /// needed as the compilation an assembly came from can be GC'ed and further requests to get that
+                /// compilation (or any of it's assemblies) may produce new assembly symbols.
+                /// </summary>
+                public readonly UnrootedSymbolSet? UnrootedSymbolSet;
```

Comment:
This is no longer weak right? Comment can be updated and the ? removed?

Reply:
yup.  will fixup.